src/server/pkg/hashtree/db.go

```diff
@@ -1591,7 +1591,7 @@ func (d *ChildCursor) V() []byte {
 
 // Next gets the next key, value pair.
 func (d *ChildCursor) Next() ([]byte, []byte) {
-	if d.k == nil {
+	if d == nil || d.c == nil {
 		return nil, nil
 	}
 	k, v := d.c.Seek(append(d.k, 1))
```

Comment:
append works with a nil array, so the check for d.k is not needed

Reply:
I think this check was to see if it is at the end of the cursor, and to not seek if it is. I don't think it is necessary to check if d or d.c is nil if we use the NewChildCursor function.